src/rrd.c

```diff
@@ -1105,23 +1175,23 @@ unsigned long long rrdset_done(RRDSET *st)
 #endif
 
 		if(unlikely(st->debug)) {
-			debug(D_RRD_STATS, "%s: last ut = %0.3Lf (last updated time)", st->name, (long double)last_ut/1000000.0);
-			debug(D_RRD_STATS, "%s: next ut = %0.3Lf (next interpolation point)", st->name, (long double)next_ut/1000000.0);
+			// FIX: Don't need 'long double' here.
+			debug(D_RRD_STATS, "%s: last ut = %0.3Lf (last updated time)", st->name, (double)last_ut/1000000.0);
+			debug(D_RRD_STATS, "%s: next ut = %0.3Lf (next interpolation point)", st->name, (double)next_ut/1000000.0);
 		}
 
 		st->last_updated.tv_sec = (time_t) (next_ut / 1000000ULL);
 		st->last_updated.tv_usec = 0;
 
+		/* scans the dimensions third time... */
```

Comment:
This is more than 3rd...

At this point it interpolates the collected values to second boundaries. The loop above is now at the beginning of each second.